### PR TITLE
Implement `Zeroize` for `SecretKey`

### LINKED ISSUE #155: Secret values not erased
#### Summary
Add a way to securely erase sensitive data from memory when types go out of scope.

#### Possible solution design or implementation
Add `Zeroize` trait, remove `Copy` trait on all structs carrying sensitive data.

#### Additional context
See [bls12_381-bls #5](https://github.com/dusk-network/bls12_381-bls/issues/5)

### PR DESCRIPTION
**Pull request by moCello**
Resolves #155

### CODE REVIEW

File: src/keys/view.rs [hunk 1]
```diff
@@ -77,12 +77,6 @@ impl ViewKey {
     }
 }
 
-impl From<SecretKey> for ViewKey {
-    fn from(sk: SecretKey) -> Self {
-        Self::from(&sk)
-    }
-}
-
```
Comment (ureeves) [inline src/keys/view.rs:85]: That's quite the catch on the by-copy. Nice

**Review by ureeves — APPROVED** [review]
Seeing all the `PublicKey::from(&sk)` make me feel there should be a `SecretKey::public` function. That said, not the purpose here. LGTM

**Review by HDauven — APPROVED** [review]
LGTM

File: src/keys/view.rs (hunk 1 above)
Comment (moCello) [reply src/keys/view.rs:85]: indeed! Will have some ripple effects downstream, but good ones!

File: src/keys/secret.rs [hunk 2]
```diff
@@ -17,7 +18,30 @@ use rand_core::{CryptoRng, RngCore};
 use subtle::{Choice, ConstantTimeEq};
 
 /// Secret pair of `a` and `b` defining a [`SecretKey`]
-#[derive(Clone, Copy, Eq, Debug)]
+///
+/// ## Safety
+///
+/// To ensure that no secret information lingers in memory after the variable
+/// goes out of scope, we advice calling `zeroize` before the variable goes out
+/// of scope.
```
Comment (Daksh14) [inline src/keys/secret.rs:26]: Would implementing `Drop` and calling `zeroize` there solve this problem?

Comment (moCello) [reply src/keys/secret.rs:26]: Sadly not. I investigated the problem in our bls signature library: https://github.com/dusk-network/bls12_381-bls/pull/6
long story short: rust still often copies values internally, so the zeroize on drop often doesn't result in actually erasing the desired values.

**Pull request merged**